pkg/storage/prunable/prunable.go

```diff
@@ -1,31 +1,50 @@
 package prunable
 
 import (
-	"github.com/iotaledger/hive.go/kvstore"
-	"github.com/iotaledger/hive.go/lo"
-	"github.com/iotaledger/iota-core/pkg/database"
+	"github.com/iotaledger/hive.go/runtime/options"
+	"github.com/iotaledger/iota-core/pkg/storage/database"
 	iotago "github.com/iotaledger/iota.go/v4"
 )
 
 const (
 	blocksPrefix byte = iota
 	rootBlocksPrefix
-	attestationsPrefix
-	ledgerStateDiffsPrefix
 )
 
 type Prunable struct {
-	Blocks           *Blocks
-	RootBlocks       *RootBlocks
-	Attestations     func(index iotago.SlotIndex) kvstore.KVStore
-	LedgerStateDiffs func(index iotago.SlotIndex) kvstore.KVStore
+	api     iotago.API
+	manager *Manager
 }
 
-func New(dbManager *database.Manager) (newPrunable *Prunable) {
+func New(dbConfig database.Config, opts ...options.Option[Manager]) *Prunable {
 	return &Prunable{
-		Blocks:           NewBlocks(dbManager, blocksPrefix),
-		RootBlocks:       NewRootBlocks(dbManager, rootBlocksPrefix),
-		Attestations:     lo.Bind([]byte{attestationsPrefix}, dbManager.Get),
-		LedgerStateDiffs: lo.Bind([]byte{ledgerStateDiffsPrefix}, dbManager.Get),
+		manager: NewManager(dbConfig, opts...),
 	}
 }
+
+func (p *Prunable) Initialize(a iotago.API) {
+	p.api = a
+}
+
+func (p *Prunable) Blocks(slot iotago.SlotIndex) *Blocks {
+	// TODO: What if the storage for this index does not exist? What about read-only access?
```

Comment:
The manager has a pruned index, so probably return an error or nil if it is too old. If it is already a finalized slot we could return a read-only instance